src/Chapter3.hs

```diff
@@ -479,6 +521,35 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
+data House = House {housePeopleNum::Int} deriving (Show)
+data MainBuilding = Church | Library deriving (Show)
+data Castle = Castle {castleName::String} | NoCastle deriving (Show)
```

Comment:
`castleName` is a partial function here. That's why records in sum types are usually discouraged. Because `castleName` will be a function of type:

```haskell
castleName :: Castle -> String
```

But it can't return a string on `NoCastle`